### PR TITLE
[DOC] remove minishift from documentation

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>
**Documentation**

Minishift / oc cluster up is not supported anymore, and has been removed from the docs.

The _Deploying_ guide has been updated as follows:

- _Alternative cluster deployment options_ section removed
- _Installing a local OpenShift cluster_ procedure removed
- `oc` and `kubectl` comparison moved to _Deployment prerequisites_ section
- _Deployment prerequisites_ section updated to remove refs to minishift
- _Installing a local Kubernetes cluster with Minikube_ section updated, and moved up a level under _Preparing for your Strimzi deployment_
- Other references to minishift updated, including metrics section

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/assembly-kafka-bridge-quickstart.adoc [hunk 1]
```diff
@@ -21,7 +21,7 @@ Use this quickstart to try out the Strimzi Kafka Bridge in your local developmen
 * Create a Kafka Bridge consumer
 * Perform basic consumer operations, such as subscribing the consumer to topics and retrieving the messages that you produced
 
-In this quickstart, HTTP requests are formatted as curl commands that you can copy and paste to your terminal. Access to a Kubernetes cluster is required; to run and manage a local Kubernetes cluster, use a tool such as Minikube, CodeReady Containers, or MiniShift.
```
Comment (PaulRMellor) [inline documentation/assemblies/assembly-kafka-bridge-quickstart.adoc:24]: I don't think we need this, as we mention local or remote Kubernetes cluster in prereqs. It was showing downstream too

**Review by scholzj — APPROVED** [review]

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 2]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:9]: _Suggested change:_

To deploy Strimzi, make sure that:

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 3]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
 
 ifdef::Downloading[]
 * A Kubernetes {KubernetesVersion} cluster is available
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:11]:
```suggestion
* A Kubernetes {KubernetesVersion} cluster is available.
```

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 4]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
...
 NOTE: Strimzi supports some features that are specific to OpenShift,
 where such integration benefits OpenShift users and there is no equivalent implementation using standard Kubernetes.
 
-ifdef::InstallationAppendix[]
 [discrete]
-== Alternatives if a Kubernetes cluster is not available
+== `oc` and `kubectl` commands
+
+The `oc` command functions as an alternative to `kubectl`.
+In almost all cases the example `kubectl` commands given in this guide can be done using `oc` simply by replacing the command name (options and arguments remain the same).
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:23]: _Suggested change:_

In almost all cases the example `kubectl` commands used in this guide can be done using `oc` simply by replacing the command name (options and arguments remain the same).

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 5]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
...
+
+In other words, instead of using:
+
+[source,shell,subs=+quotes]
+kubectl apply -f _your-file_
 
-If you do not have access to a Kubernetes cluster, as an alternative you can try installing Strimzi with:
+when using OpenShift you can use
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:30]:
```suggestion
when using OpenShift you can use:
```

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 6]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
...
 
-* xref:deploy-kubernetes-{context}[_Minikube_]
-* xref:deploy-openshift-{context}[_Minishift_]
+[source,shell,subs=+quotes]
+oc apply -f _your-file_
 
-endif::InstallationAppendix[]
+NOTE: As an exception to this general rule, `oc` uses `oc adm` subcommands for _cluster management_,
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:35]: _Suggested change:_

NOTE: As an exception to this general rule, `oc` uses `oc adm` subcommands for _cluster management_ functionality, whereas `kubectl` does not make this distinction.

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 7]
```diff
@@ -9,24 +9,38 @@ To deploy Strimzi, make sure:
...
+while `kubectl` does not make such a distinction.
+For example, the `oc` equivalent of `kubectl taint` is `oc adm taint`.
+
+endif::Downloading[]
+ifndef::Downloading[]
+* An OpenShift {OpenShiftVersion} cluster is available
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:41]:
```suggestion
* An OpenShift {OpenShiftVersion} cluster is available.
```

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Looks good. I made a few comments.

**Review by laidan6000 — APPROVED** [review]
Changes look good.

**Pull request merged**